### PR TITLE
Added new e2e test to initiate WCCOM connection

### LINKED ISSUE #29414: Write new e2e test: Merchant - Extensions - Can initiate WCCOM connection
Write a new e2e test that covers the following flow:

- Merchant - Extensions - Can initiate WCCOM connection

### PR DESCRIPTION
**Pull request by veljkho**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #29414

### How to test the changes in this Pull Request:

- npx wc-e2e test:e2e ./tests/e2e/specs/wp-admin/test-extensions-connect-wccom.js

 PASS  ../specs/wp-admin/test-extensions-connect-wccom.js (12.734s)
  Merchant > Initiate WCCOM Connection
    ✓ can initiate WCCOM connection (6676ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        12.78s

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: tests/e2e/core-tests/specs/index.js [hunk 1]
```diff
@@ -154,4 +156,5 @@ module.exports = {
 	runCartRedirectionTest,
 	runMyAccountCreateAccountTest,
 	runOrderEmailReceivingTest,
+	runInitiateWccomConnectionTest,
```
Comment (rrennick) [inline tests/e2e/core-tests/specs/index.js:159]: This is having the test run twice. The api tests are ones that test the local install's rest api.

Comment (rrennick) [reply tests/e2e/core-tests/specs/index.js:159]: Sorry, please ignore this comment.

**Comment by rrennick** [comment]
@veljkho I merged this with `trunk` to test locally to ensure it passed with all the changes we had made over the last couple weeks. 

Since we renamed the files in `tests/e2e/specs/wp-admin`, can you rename `test-extensions-connect-wccom.js` to `wccom-connect.test.js` so this test is included automatically?

**Comment by veljkho** [comment]
> @veljkho I merged this with `trunk` to test locally to ensure it passed with all the changes we had made over the last couple weeks.
> 
> Since we renamed the files in `tests/e2e/specs/wp-admin`, can you rename `test-extensions-connect-wccom.js` to `wccom-connect.test.js` so this test is included automatically?

Done. This is now ready for a review.

**Review by rrennick — APPROVED** [review]
@veljkho Nice work. This tested great for me locally.

**Pull request merged**